community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -487,6 +487,7 @@ public void start() throws IOException
             dependencies.satisfyDependency( storageEngine.storeReadLayer() );
             dependencies.satisfyDependency( logEntryReader );
             dependencies.satisfyDependency( storageEngine );
+            dependencies.satisfyDependency( transactionLogModule.checkPointing() );
```

Comment:
i think you need this at all, for example because  https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java#L464
hack, and there is for example CheckpointerSupplier that rely on checkpointer to be registered